extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/UnremovableBeanBuildItem.java

```diff
@@ -58,6 +59,46 @@ public boolean test(BeanInfo bean) {
 
     }
 
+    public static class BeanDotNameExclusion implements Predicate<BeanInfo> {
```

Comment:
This requires a review from @mkouba .

Reply:
This class should be probably named `BeanTypeExclusion` because that's what it does - excludes from removal if a bean has a specific type in its set of bean types. The same applies to the `BeanDotNamesExclusion` class.